articles/cross_compilation_build_tools_design.md

```diff
@@ -0,0 +1,248 @@
+---
+layout: default
+title: Cross-compiling ROS 2 packages
+abstract:
+  This article is a design proposal for extending the build tools for ROS 2 to
+  simplify cross-compiling ROS 2 packages for platforms currently not supported
```

Comment:
I would suggest that we define what a platform is, e.g. in the automotive world a platform is a tupple of {ECU:OS:Framework}. So for instance an example of a platform would be: {Renesas R-Car H3:QNX:ROS2 C}.
Note that Renesas R-Car H3 implicitly implies that this is an `aarch64` architecture for computer processor.

Since ROS 2 depends greatly on an underlying middle-ware it would be good to even include that into the tupple, so {ECU:OS:Middle-ware:Framework}.

Reply:
Perhaps we can have a generic part and a platform-specific set of use cases?

Reply:
I agree with the proposal to add the ROS distro and rmw implementation. For the ECU I propose starting with a `generic_armhf`, and then adding some specific ECUs as @gbiggs suggests. So we for example we could have `generic_armhf-ubuntu_bionic-fastrtps-crystal` as an example  platform identifier.